src/main/cpp/FfmpegJavaAvPlaybackPipline.cpp

```diff
@@ -0,0 +1,368 @@
+	}
+
+	pJavaPlayback->set_balance(fBalance);
+
+	return ERROR_NONE;
+}
+
+uint32_t FfmpegJavaAvPlaybackPipline::GetBalance(float* pfBalance) {
+	// TODO(fraudies): Not sure how to wire this
+	if (pJavaPlayback == nullptr) {
+		return ERROR_PLAYER_NULL;
+	}
+
+	*pfBalance = pJavaPlayback->get_balance();
+
+	return ERROR_NONE;
+}
+
+uint32_t FfmpegJavaAvPlaybackPipline::SetAudioSyncDelay(long lMillis) {
+	// TODO(fraudies): Implement this
+	if (pJavaPlayback == nullptr) {
+		return ERROR_PLAYER_NULL;
+	}
+
+	pJavaPlayback->set_audioSyncDelay(lMillis);
+
+	return ERROR_NONE; // no error
+}
+
+uint32_t FfmpegJavaAvPlaybackPipline::GetAudioSyncDelay(long* plMillis) {
+	// TODO(fraudies): Implement this
+	if (pJavaPlayback == nullptr) {
+		return ERROR_PLAYER_NULL;
+	}
+
+	*plMillis = pJavaPlayback->get_audioSyncDelay();
+
+	return ERROR_NONE; // no error
+}
+
+uint32_t FfmpegJavaAvPlaybackPipline::HasAudioData(bool* bAudioData) const {
+	if (pJavaPlayback == nullptr) {
+		return ERROR_PLAYER_NULL;
+	}
+	// TODO(fraudies): Implement this
+	*bAudioData = pJavaPlayback->has_audio_data();
+
+	return ERROR_NONE;
+}
+
+uint32_t FfmpegJavaAvPlaybackPipline::HasImageData(bool* bImageData) const {
+	if (pJavaPlayback == nullptr) {
+		return ERROR_PLAYER_NULL;
+	}
+	// TODO(fraudies): Implement this
+	*bImageData = pJavaPlayback->has_image_data();
+
+	return ERROR_NONE;
+}
+
+uint32_t FfmpegJavaAvPlaybackPipline::GetImageWidth(int* width) const {
+	if (pJavaPlayback == nullptr) {
+		return ERROR_PLAYER_NULL;
+	}
+	// TODO(fraudies): Implement this
+	*width = pJavaPlayback->get_image_width();
+
+	return ERROR_NONE;
+}
+
+uint32_t FfmpegJavaAvPlaybackPipline::GetImageHeight(int* iHeight) const {
+	if (pJavaPlayback == nullptr) {
+		return ERROR_PLAYER_NULL;
+	}
+	// TODO(fraudies): Implement this
+	*iHeight = pJavaPlayback->get_image_height();
+
+	return ERROR_NONE;
+}
+
+uint32_t FfmpegJavaAvPlaybackPipline::GetAudioFormat(AudioFormat* pAudioFormat) const {
+	if (pJavaPlayback == nullptr) {
+		return ERROR_PLAYER_NULL;
+	}
+	// TODO(fraudies): Implement this
+	*pAudioFormat = pJavaPlayback->get_audio_format();
+
+	return ERROR_NONE;
+}
+
+uint32_t FfmpegJavaAvPlaybackPipline::GetPixelFormat(PixelFormat* pPixelFormat) const {
+	if (pJavaPlayback == nullptr) {
+		return ERROR_PLAYER_NULL;
+	}
+	// TODO(fraudies): Implement the pixel format for the ouput
+	*pPixelFormat = pJavaPlayback->get_pixel_format();
+
+	return ERROR_NONE;
+}
+
+uint32_t FfmpegJavaAvPlaybackPipline::GetImageBuffer(uint8_t** ppImageBuffer) {
+	if (pJavaPlayback == nullptr) {
+		return ERROR_PLAYER_NULL;
+	}
+	// TODO(fraudies): Implement the image buffer data
+	return ERROR_NONE;
+}
+
+uint32_t FfmpegJavaAvPlaybackPipline::GetAudioBuffer(uint8_t** ppAudioBuffer) {
+	if (pJavaPlayback == nullptr) {
+		return ERROR_PLAYER_NULL;
+	}
+	// TODO(fraudies): Implement the audio buffer data
+	return ERROR_NONE;
+}
+
+void FfmpegJavaAvPlaybackPipline::UpdatePlayerState(PlayerState newState) {
```

Comment:
Add TODO: Push this logic into a super-class that is common the SDL / Java implementation

Reply:
Added.